tests/testFlagHandler.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2013 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import unittest
+
+import lsst.utils.tests
+import lsst.meas.base
+import lsst.meas.base.tests
+import lsst.afw.table
+from lsst.meas.base.baseLib import MeasurementError
+from lsst.meas.base import FlagDefinition, FlagDefinitionVector, FlagHandler
+from lsst.meas.base.tests import (AlgorithmTestCase)
+
+class FlagHandlerTestCase(AlgorithmTestCase):
+    
+    def testFlagHandler(self):
+        control = lsst.meas.base.GaussianCentroidControl()
+        alg = lsst.meas.base.GaussianCentroidAlgorithm
+        schema = lsst.afw.table.SourceTable.makeMinimalSchema()
+        plugin = alg(control, 'test', schema)
+        cat = lsst.afw.table.SourceCatalog(schema)
+        schema = cat.getSchema()
+        subSchema = schema["test"]
+
+        # This is a FlagDefinition structure like a plugin might have
+        FAILURE = 0
+        FIRST = 1
+        SECOND = 2
+        flagDefs = [ FlagDefinition("General Failure", "general failure error"),
+            FlagDefinition("1st error", "this is the first failure type"),
+            FlagDefinition("2nd error", "this is the second failure type")
+        ]
+        fh = FlagHandler.addFields(schema, "test",
+            FlagDefinitionVector(flagDefs))
+
+        # Check to be sure that the FlagHandler was correctly initialized
+        for index, flagDef in enumerate(flagDefs):
+           assert(flagDef.name == flagDefs[index].name)
+           assert(flagDef.doc == flagDefs[index].doc)
```

Comment:
This doesn't actually test what you say it does in the comment, you are actually just comparing an object to itself, as can be seen with id(). I think you are trying to compare to fh.getDefinition[index].name. Please fix the test.